### PR TITLE
Update media UI

### PR DESCRIPTION
**Pull request by breadadams**
+ Update media progress bar component to handle interactions for seeking
+ Add spinner component for while media is buffering
+ Misc. refactoring

#### To-do:
- [ ] Add new classnames to docs

### CODE REVIEW

**Comment by Kikobeats** [comment]
What do you think about show the spinner at play/pause place?

File: packages/react/src/components/Card/CardMedia/Controls/MediaButton.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ const MediaButton = styled('div')`
   backface-visibility: hidden;
   filter: drop-shadow(0 2px 4px rgba(0, 0, 0, 0.3));
   transition: transform ${transition.short};
+  will-change: transform;
```
Comment (Kikobeats) [inline packages/react/src/components/Card/CardMedia/Controls/MediaButton.js]: this probably need to be added around all the code! I need to review it to ensure `will-change` is there every time `transition` is defined 🙂

**Comment by breadadams** [comment]
@Kikobeats 

> What do you think about showing the spinner at play/pause place?

I tried this out while adding the spinner, it looks a bit odd in the center on the large card. But also, that means hiding the seek + play/pause buttons while the media is buffering - when you _should_ be able to still interact with the player (I've took inspiration from other players eg. youtube).

> Another thing: Do you think it could be more or less easy to add a "magnified" effect every time you are trying to jump to a section in the progress bar?

There's already a css `scale` transform on the progress bar on `:hover`, maybe we can increase it?

File: packages/react/src/components/Card/CardMedia/Controls/ProgressBar/index.js [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+      }
+    }
+  }
+
+  return {}
+})`
+  background: transparent;
+  border-radius: 9999px;
+  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.3);
+  border: 1px solid rgba(255, 255, 255, 0.75);
```
Comment (Kikobeats) [inline packages/react/src/components/Card/CardMedia/Controls/ProgressBar/index.js]:
```suggestion
```

Comment (Kikobeats) [reply packages/react/src/components/Card/CardMedia/Controls/ProgressBar/index.js]: Just to try a clean UI version 🙂

File: packages/react/src/theme.js [hunk 3]
```diff
@@ -10,10 +10,20 @@ export const animation = {
   long: 'cubic-bezier(.4, 0, .2, 1)'
 }
 
+const createTransition = (properties, s) => {
+  if (Array.isArray(properties)) {
+    const suffix = `${speed[s]} ${animation[s]}`
+
```
Comment (Kikobeats) [inline packages/react/src/theme.js]:
```suggestion
```

File: packages/react/src/theme.js [hunk 4]
```diff
@@ -10,10 +10,20 @@ export const animation = {
...
+    return properties.map(property => `${property} ${suffix}`).join(', ')
+  }
+
```
Comment (Kikobeats) [inline packages/react/src/theme.js]:
```suggestion
```

File: packages/react/src/theme.js [hunk 5]
```diff
@@ -10,10 +10,20 @@ export const animation = {
...
+  return ''
+}
+
 export const transition = {
-  short: `${speed.short} ${animation.short}`,
-  medium: `${speed.medium} ${animation.medium}`,
-  long: `${speed.long} ${animation.long}`
+  short: (...p) => createTransition(p, 'short'),
```
Comment (Kikobeats) [inline packages/react/src/theme.js]:
```suggestion
  short: (...properties) => createTransition(properties, 'short'),
```

File: packages/react/src/theme.js [hunk 6]
```diff
@@ -10,10 +10,20 @@ export const animation = {
...
+  medium: (...p) => createTransition(p, 'medium'),
```
Comment (Kikobeats) [inline packages/react/src/theme.js]:
```suggestion
  medium: (...properties) => createTransition(properties, 'medium'),
```

File: packages/react/src/theme.js [hunk 7]
```diff
@@ -10,10 +10,20 @@ export const animation = {
...
+  long: (...p) => createTransition(p, 'long')
```
Comment (Kikobeats) [inline packages/react/src/theme.js]:
```suggestion
  long: (...properties) => createTransition(properties, 'long')
```

File: packages/react/src/theme.js [hunk 8]
```diff
@@ -10,10 +10,20 @@ export const animation = {
   long: 'cubic-bezier(.4, 0, .2, 1)'
 }
 
+const createTransition = (properties, s) => {
+  if (Array.isArray(properties)) {
```
Comment (Kikobeats) [inline packages/react/src/theme.js]: what's the point of checking if `properties` is an array if always it's passed as an array?

**Comment by Kikobeats** [comment]
> There's already a css scale transform on the progress bar on :hover, maybe we can increase it?

oh, it's true. I think if we increment it a bit could be nice, making more evident the bar interaction. Although probably that means we need to increment a bit the "bottom" value of the bottom elements, there isn't too much space between them

File: packages/react/src/utils/index.js [hunk 9]
```diff
@@ -86,6 +86,29 @@ export const imageProxy = url =>
 
 export const isLazySupported = !isSSR && 'IntersectionObserver' in window
 
+export const formatSeconds = secs => {
```
Comment (Kikobeats) [inline packages/react/src/utils/index.js]: maybe we can use a browser API to have the same effect in less code?

File: packages/react/src/theme.js (hunk 8 above)
Comment (breadadams) [reply packages/react/src/theme.js]: Yup! You're right, it was from an earlier version I did where I was passing either an array or a string, until I realized I could just destructure the parameters 😅

File: packages/react/src/utils/index.js (hunk 9 above)
Comment (breadadams) [reply packages/react/src/utils/index.js]: Hmm what do you have in mind @Kikobeats?

**Pull request merged**